vnext/src/IntegrationTests/XHRTest.js

```diff
@@ -31,78 +31,83 @@ class XHRTest extends React.Component<{}, Object> {
-    }
-    xhr.onload = () => {
-      this.setState({downloading: false});
-      if (xhr.status === 200) {
-        let responseType = `Response is a string, ${
-          xhr.response.length
-        } characters long.`;
-        if (xhr.response instanceof ArrayBuffer) {
-          responseType = `Response is an ArrayBuffer, ${
-            xhr.response.byteLength
-          } bytes long.`;
+      const onreadystatechange = () => {
+        if (xhr.readyState === xhr.HEADERS_RECEIVED) {
+          const contentLength = parseInt(
+            xhr.getResponseHeader('Content-Length'),
+            10,
+          );
+          this.setState({
+            contentLength,
+            responseLength: 0,
+          });
+        } else if (xhr.readyState === xhr.LOADING && xhr.response) {
+          this.setState({
+            responseLength: xhr.response.length,
+          });
         }
-        Alert.alert('Download complete!', responseType);
-      } else if (xhr.status !== 0) {
-        Alert.alert(
-          'Error',
-          `Server returned HTTP status of ${xhr.status}: ${xhr.responseText}`,
-        );
-      } else {
-        Alert.alert('Error', xhr.responseText);
+      };
+
+      if (this.state.readystateHandler) {
+        xhr.onreadystatechange = onreadystatechange;
+      }
+      //ISS:2306365 - Uncomment when native module is complete.
+      //if (this.state.progressHandler) {
+      //  xhr.onprogress = onprogress;
+      //}
+      if (this.state.arraybuffer) {
+        xhr.responseType = 'arraybuffer';
       }
-    };
-    xhr.open('GET', 'http://aleph.gutenberg.org/cache/epub/100/pg100.txt.utf8');
-    // Avoid gzip so we can actually show progress
-    xhr.setRequestHeader('Accept-Encoding', '');
-    xhr.send();
+      xhr.onload = () => {
+        this.setState({downloading: false});
+        if (xhr.status === 200) {
+          let responseType = `Response is a string, ${
+            xhr.response.length
+          } characters long.`;
+          if (xhr.response instanceof ArrayBuffer) {
+            responseType = `Response is an ArrayBuffer, ${
+              xhr.response.byteLength
+            } bytes long.`;
+          }
+          Alert.alert('Download complete!', responseType);
+        } else if (xhr.status !== 0) {
+          Alert.alert(
+            'Error',
+            `Server returned HTTP status of ${xhr.status}: ${xhr.responseText}`,
+          );
+        } else {
+          Alert.alert('Error', xhr.responseText);
+        }
+      };
+      xhr.open('GET', 'https://github.com/microsoft/react-native-windows/raw/master/.github/hero2.png');
+      // Avoid gzip so we can actually show progress
+      xhr.setRequestHeader('Accept-Encoding', '');
+      xhr.send();
+
+      this.setState({downloading: true});
+    });
 
-    this.setState({downloading: true});
   };
 
   componentDidMount() {
     try {
-      this._download();
+      async () => await this._download();
```

Comment:
As is, this doesn't actually call this._download. This defines an anonymous async function that isn't called.

You're not going to be able to await an async function in componentDidMount/use Effect. You'll probably just want to mark the test passed at the point in the test sequence where we know it's done.

Reply:
Alternatively, you could expand the async function, and Mark the test passed/failed within it.

Reply:
> As is, this doesn't actually call this._download. This defines an anonymous async function that isn't called.
> 
> You're not going to be able to await an async function in componentDidMount/useEffect. You'll probably just want to mark the test passed at the point in the test sequence where we know it's done.

I think I should be able to, but I did skip invoking the promise: ` (async() => awiat this._downloadI())();`
I'll give it one more try, else add a follow-up task.

Reply:
The easier path if you're wanting to adapt this code is to probably just move the `TestModule.markTestPassed()` to a different place and not try to incorporate the async function.

We do have a wrapper to allow writing tests in async functions in the new ITs, but it's a little bit tricky. See the below in `TestDefinition.tsx` if you're interested (there are examples of calls to `functionTest` in `SampleTests.tsx`.

```
/**
 * Create a root component which runs the passed test function
 */
function createImperativeTestComponent(fn: TestFunction): React.FC {
  return () => {
    useEffect(() => {
      (async () => {
        try {
          const testResult = fn();
          if (testResult instanceof Promise) {
            await testResult;
          }
        } catch (ex) {
          // Need to force a redbox instead of YellowBox for unhandled
          // promise rejection
          ErrorUtils.getGlobalHandler()(ex);
          return;
        }

        TestModule.markTestPassed(true);
      })();
    });

    return <View />;
  };
}
```